src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveReferencesToAliasesPassTest.php

```diff
@@ -123,6 +123,28 @@ public function testDeprecationNoticeWhenReferencedByDefinition()
         $this->process($container);
     }
 
+    public function testNoDeprecationNoticeWhenReferencedByDeprecatedAlias()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('foo', 'stdClass');
+
+        $aliasDeprecated = new Alias('foo');
+        $aliasDeprecated->setDeprecated(true);
+        $container->setAlias('deprecated_foo_alias', $aliasDeprecated);
+
+        $alias = new Alias('deprecated_foo_alias');
+        $alias->setDeprecated(true);
+        $container->setAlias('alias', $alias);
+
+        $alias = new Alias('deprecated_foo_alias');
+        $alias->setDeprecated(true);
+        $container->setAlias('alias', $alias);
```

Comment:
this is useless as it overrides the one just above. I think you duplicate the block by mistake.

Reply:
right, fixed